app/services/companies_list_updator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CompaniesListUpdator < ApplicationService
+  def call
+    companies = GPW::Client.new.fetch_companies_list
+
+    companies.each do |company|
+      Company.find_by(isin: company.fetch(:isin))&.update!(
+        name: company.fetch(:name),
+      )
+      Company.find_or_create_by!(company)
```

Comment:
You are doing find twice. Also it seems hard to understand what is going on.

Maybe something like this would work:

```ruby
def upsert_company(attrs)
  company = Company.find_by(isin: company.fetch(:isin))

  if company
    company.update!(name: attrs.fetch(:name))
  else
    Company.create!(attrs)
  end
end
```

Reply:
Yes, this is avoid finding company twice